Gem/PythonTests/Automated/test_suites/periodic/ModelHotReloading_test_case.py

```diff
@@ -0,0 +1,261 @@
+    materialLabelDict = {modelMaterialOverrideLod:lodMaterialList, 0:lodMaterialList}    
+    validate_material_slot_labels(myEntityId, materialLabelDict)
+
+    # Test that increasing the lod count functions
+    if not modelReloadHelper.copy_file_and_wait_for_on_model_ready(myEntityId, 'Objects/ModelHotReload/sphere_5lods.fbx'):
+        general.log("OnModelReady never happened - sphere_5lods.fbx")
+    
+    # The model material overrides have 5 slots, each individual lod only has 1 slot
+    materialLabelDict = {modelMaterialOverrideLod:["lambert0", "lambert3", "lambert4", "lambert5", "lambert6"], 0:["lambert0"], 1:["lambert3"], 2:["lambert4"], 3:["lambert5"], 4:["lambert6"]}
+    validate_material_slot_labels(myEntityId, materialLabelDict)
+
+    # Test that adding a vertex stream and removing lods functions
+    if not modelReloadHelper.copy_file_and_wait_for_on_model_ready(myEntityId, 'Objects/ModelHotReload/vertexcolor.fbx'):
+        general.log("OnModelReady never happened - vertexcolor.fbx")
+    # Use the presence of a single material slot in the material component to validate the m
+    lodMaterialList = ["Material"]
+    materialLabelDict = {modelMaterialOverrideLod:lodMaterialList, 0:lodMaterialList}
+    validate_material_slot_labels(myEntityId, materialLabelDict)
+    
+    # Future steps for other test cases
+    # apply a material override to two of the materials
+    # apply a property override to one of the materials
+    
+    # Default material not properly being cleared
+    # - apply default material on one model
+    #  - reload to different model (sphere5_lods)
+    # - apply material slot overrides
+    # - clear the default material assignment (this might have been done before reloading the mesh, or without reloading at all, just assigning a new mesh) (it might have been done before or after applying the slot overrides)
+    # - expected: slots without overrides use their default material
+    # - actual: slots without overrides use the old default material
+
+    # remove one of the materials
+    # change some faces to use the same material as one of the already overriden slots
+    # also change some faces to use one of the materials that has the default applied in the material component
+    # also change some faces to use a newly added material
+
+    # enable lod material override
+    
+    # Repeat with the actor component
+
+    
+    # Close the Editor to end the test.
+    helper.close_editor()
+
+def print_material_slot_labels(entityId):
+    # Helper function useful while writing/modifying the test that will output the available materials slots
+    general.log("Printing Material Slot AssignmentIds and Labels")
+    materialAssignmentMap = azlmbr.render.MaterialComponentRequestBus(azlmbr.bus.Event, 'GetOriginalMaterialAssignments', entityId)
+    for assignmentId in materialAssignmentMap:
+        general.log("  AssignementId (slotId:lod): {0}".format(assignmentId.ToString()))
```

Comment:
f-string is generally preferred to .format 
```suggestion
        general.log(f"  AssignementId (slotId:lod): {assignmentId}")
```

Reply:
I have to use assignmentId.ToString() here, but that works. Did not know about f-string, it is way better!